src/System.Runtime.Extensions/tests/System/IO/PathTestsBase.cs

```diff
@@ -201,235 +201,253 @@ public partial class PathTestsBase
+
+            { @"foo", @"\\.\UNC\LOCALHOST\share8\test.txt.~SS", @"\\.\UNC\LOCALHOST\share8\test.txt.~SS\foo" },
+            { @"foo", @"\\.\UNC\LOCALHOST\share9", @"\\.\UNC\LOCALHOST\share9\foo" },
+            { @"foo", @"\\.\UNC\LOCALHOST\shareA\dir", @"\\.\UNC\LOCALHOST\shareA\dir\foo" },
+            { @". \foo", @"\\.\UNC\LOCALHOST\shareB\", @"\\.\UNC\LOCALHOST\shareB\. \foo" },
+            { @".. \foo", @"\\.\UNC\LOCALHOST\shareC\", @"\\.\UNC\LOCALHOST\shareC\.. \foo" },
+            { @"    \foo", @"\\.\UNC\LOCALHOST\shareD\", @"\\.\UNC\LOCALHOST\shareD\    \foo" },
+
+            { "foo", @"\\.\UNC\LOCALHOST\  shareE\", @"\\.\UNC\LOCALHOST\  shareE\foo" },
+            { "foo", @"\\.\UNC\LOCALHOST\shareF\test.txt.~SS", @"\\.\UNC\LOCALHOST\shareF\test.txt.~SS\foo" },
+            { "foo", @"\\.\UNC\LOCALHOST\shareG", @"\\.\UNC\LOCALHOST\shareG\foo" },
+            { "foo", @"\\.\UNC\LOCALHOST\shareH\dir", @"\\.\UNC\LOCALHOST\shareH\dir\foo" },
+            { "foo", @"\\.\UNC\LOCALHOST\shareK\", @"\\.\UNC\LOCALHOST\shareK\foo" },
+            { "foo", @"\\.\UNC\LOCALHOST\  shareL\", @"\\.\UNC\LOCALHOST\  shareL\foo" },
+
+            // UNC path starting with \\
+            { @"foo", @"\\server1", @"\\server1\foo" },
+            { @"\foo", @"\\server2", @"\\server2\foo" },
+            { @"foo", @"\\server3\", @"\\server3\foo" },
+            { @"foo", @"\\server3\\", @"\\server3\foo" },
+            { @"..\foo", @"\\server4", @"\\server4\..\foo" },
+            { @".\foo", @"\\server5\share", @"\\server5\share\foo" },
+            { @"..\foo", @"\\server6\share", @"\\server6\share\foo" },
+            { @"\foo", @"\\a\b\\", @"\\a\b\foo" },
+
+            { @"foo", @"\\LOCALHOST\share8\test.txt.~SS", @"\\LOCALHOST\share8\test.txt.~SS\foo" },
+            { @"foo", @"\\LOCALHOST\share9", @"\\LOCALHOST\share9\foo" },
+            { @"foo", @"\\LOCALHOST\shareA\dir", @"\\LOCALHOST\shareA\dir\foo" },
+            { @". \foo", @"\\LOCALHOST\shareB\", @"\\LOCALHOST\shareB\. \foo" },
+            { @".. \foo", @"\\LOCALHOST\shareC\", @"\\LOCALHOST\shareC\.. \foo" },
+            { @"    \foo", @"\\LOCALHOST\shareD\", @"\\LOCALHOST\shareD\    \foo" },
+
+            { "foo", @"\\LOCALHOST\share1", @"\\LOCALHOST\share1\foo" },
+            { "foo", @"\\LOCALHOST\share2", @"\\LOCALHOST\share2\foo" },
+            { "foo", @"\\LOCALHOST\share3\dir", @"\\LOCALHOST\share3\dir\foo" },
+            { "foo", @"\\LOCALHOST\share4", @"\\LOCALHOST\share4\foo" },
+            { "foo", @"\\LOCALHOST\share5", @"\\LOCALHOST\share5\foo" },
+            { "foo", @"\\LOCALHOST\share6\", @"\\LOCALHOST\share6\foo" },
+            { "foo", @"\\LOCALHOST\  share7\", @"\\LOCALHOST\  share7\foo" }
+        };
 
-        public static TheoryData<string, string, string> GetFullPath_BasePath_BasicExpansions_TestData_Unix()
+        public static TheoryData<string, string, string> GetFullPath_Windows_FullyQualified => new TheoryData<string, string, string>
```

Comment:
These aren't FullyQualified. They're all sorts of paths: fully and not fully qualified, rooted and unrooted...